### PR TITLE
Integrate nuttx and pysimCoder

### PR DESCRIPTION
**Pull request by robertobucher**
## Summary
Correction of two files under arch/arm/src/stm32f7 to avoid compilation errors by PWM multichannels and qencoder

Imported files for CAN and qencoder in STM32F7 (imported from STM32L4)
The nucleo-144 device under stm32f7 has been prepared to handle the following devices with a new configuration "nucleo-144:f746-pysim"

4 GPIO IN
4 GPIO OUT
4 PWM channels
3 ADC channels
1 encoder
CAN bus

## Impact
arm -> stm32f7

## Testing
The configuration and the drivers have been testet with a NUCLEO-F746ZG board

### CODE REVIEW

**Comment by acassis** [comment]
Hi @robertobucher there are some Coding Style issue in the board (some were already there because this board config is older than our CI tests)
Please run:
$ ./tools/checkpatch.sh -f boards/arm/stm32f7/nucleo-144/include/board.h

It will show all lines with the issues

File: boards/arm/stm32f7/nucleo-144/src/stm32_qencoder.c [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/****************************************************************************
+ * boards/arm/stm32f7/nucleo-144/src/stm32_qencoder.c
+ *
+ *   Copyright (C) 2012 Gregory Nutt. All rights reserved.
```
Comment (xiaoxiang781216) [inline boards/arm/stm32f7/nucleo-144/src/stm32_qencoder.c:4]: need change to apache copyright

File: boards/arm/stm32f7/nucleo-144/src/stm32_can.c [hunk 2]
```diff
@@ -0,0 +1,134 @@
+/****************************************************************************
+ * boards/arm/stm32f7/nucleo-144/src/stm32_can.c
+ *
+ *   Copyright (C) 2018 Gregory Nutt. All rights reserved.
```
Comment (xiaoxiang781216) [inline boards/arm/stm32f7/nucleo-144/src/stm32_can.c:4]: need change to apache copyright

File: boards/arm/stm32f7/nucleo-144/src/nucleo-144.h [hunk 3]
```diff
@@ -181,15 +181,32 @@
 
 /* GPIO pins used by the GPIO Subsystem */
 
-#define BOARD_NGPIOIN     1 /* Amount of GPIO Input pins */
-#define BOARD_NGPIOOUT    1 /* Amount of GPIO Output pins */
+#define BOARD_NGPIOIN        4    /* Amount of GPIO Input pins */
```
Comment (xiaoxiang781216) [inline boards/arm/stm32f7/nucleo-144/src/nucleo-144.h:184]: align the code with other line

File: boards/arm/stm32f7/nucleo-144/configs/f746-pysim/defconfig [hunk 4]
```diff
@@ -0,0 +1,1714 @@
+#
+# Automatically generated file; DO NOT EDIT.
+# NuttX/x86_64 Configuration
+#
+
+#
+# Build Setup
```
Comment (xiaoxiang781216) [inline boards/arm/stm32f7/nucleo-144/configs/f746-pysim/defconfig:7]: need normialize by:
./tools/refresh.sh --silent nucleo-144:f746-pysim

File: boards/arm/stm32f7/nucleo-144/configs/f746-pysim/Make.defs [hunk 5]
```diff
@@ -0,0 +1,80 @@
+############################################################################
+# boards/arm/stm32f7/nucleo-144/configs/f746-pysim/Make.defs
+#
+#   Copyright (C) 2016 Gregory Nutt. All rights reserved.
+#   Authors: Gregory Nutt <gnutt@nuttx.org>
+#            Mark Olsson <post@markolsson.se>
+#            David Sidrane <david_s5@nscdg.com>
+#
+# Redistribution and use in source and binary forms, with or without
```
Comment (xiaoxiang781216) [inline boards/arm/stm32f7/nucleo-144/configs/f746-pysim/Make.defs:9]: need change to apache copyright

**Comment by xiaoxiang781216** [comment]
@robertobucher it's better to fix the style and copyright issue in the corresponding patch, to avoid the issue exist in the individual patch.

**Comment by acassis** [comment]
Hi @robertobucher you didn't need to close the PR, you can use "git rebase -1 HEAD~N" to squash the commits (where N is the number of commits you want to modify)

**Pull request closed without merging**